packages/react-native-gesture-handler/apple/RNGestureHandlerEvents.mm

```diff
@@ -213,21 +214,25 @@ + (NSString *)moduleDotMethod
 
 - (NSArray *)arguments
 {
-  NSMutableDictionary *body = [NSMutableDictionary dictionaryWithDictionary:_extraData.data];
-  [body setObject:_viewTag forKey:@"target"];
-  [body setObject:_handlerTag forKey:@"handlerTag"];
-  [body setObject:@(_state) forKey:@"state"];
-  return @[ self.viewTag, @"onGestureHandlerEvent", body ];
+  if (_actionType == RNGestureHandlerActionTypeNativeDetectorAnimatedEvent) {
+    NSMutableDictionary *body = [[NSMutableDictionary alloc] init];
+    [body setObject:_viewTag forKey:@"target"];
+    [body setObject:_handlerTag forKey:@"handlerTag"];
+    [body setObject:@(_state) forKey:@"state"];
```

Comment:
Not a strong opinion, but we could use ternary to initialize `body` and those 3 `setObject` would be the same for both.

On the other hand, it still requires `if` for returning value, so do as you think is the best 😅